src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -4453,99 +4476,23 @@ namespace EvaporativeCoolers {
-                    Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_1_Index).InletNode).MassFlowRateMaxAvail = 0.0;
-                    Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_1_Index).OutletNode).MassFlowRate = 0.0;
-                    Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_1_Index).OutletNode).MassFlowRateMaxAvail = 0.0;
-
-                    if (ZoneEvapUnit(UnitNum).EvapCooler_2_Index > 0) {
-                        Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_2_Index).InletNode).MassFlowRate = 0.0;
-                        Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_2_Index).InletNode).MassFlowRateMaxAvail = 0.0;
-                        Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_2_Index).OutletNode).MassFlowRate = 0.0;
-                        Node(EvapCond(ZoneEvapUnit(UnitNum).EvapCooler_2_Index).OutletNode).MassFlowRateMaxAvail = 0.0;
-                    }
 
-                    if (ZoneEvapUnit(UnitNum).UnitReliefNodeNum > 0) {
-                        Node(ZoneEvapUnit(UnitNum).UnitReliefNodeNum).MassFlowRate = 0.0;
-                        Node(ZoneEvapUnit(UnitNum).UnitReliefNodeNum).MassFlowRateMaxAvail = 0.0;
-                    }
-                    if (ZoneEvapUnit(UnitNum).FanLocation == BlowThruFan) {
-                        if (ZoneEvapUnit(UnitNum).FanType_Num != DataHVACGlobals::FanType_SystemModelObject) {
-                            Fans::SimulateFanComponents(
-                                ZoneEvapUnit(UnitNum).FanName, false, ZoneEvapUnit(UnitNum).FanIndex, _, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
-                        } else {
-                            HVACFan::fanObjs[ZoneEvapUnit(UnitNum).FanIndex]->simulate(_, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
-                        }
-                    }
-
-                    if (ZoneEvapUnit(UnitNum).EvapCooler_1_AvailStatus) {
-                        SimEvapCooler(ZoneEvapUnit(UnitNum).EvapCooler_1_Name, ZoneEvapUnit(UnitNum).EvapCooler_1_Index);
-                    }
-
-                    if ((ZoneEvapUnit(UnitNum).EvapCooler_2_Index > 0) && ZoneEvapUnit(UnitNum).EvapCooler_2_AvailStatus) {
-                        SimEvapCooler(ZoneEvapUnit(UnitNum).EvapCooler_2_Name, ZoneEvapUnit(UnitNum).EvapCooler_2_Index);
-                    }
-                    if (ZoneEvapUnit(UnitNum).FanLocation == DrawThruFan) {
-                        if (ZoneEvapUnit(UnitNum).FanType_Num != DataHVACGlobals::FanType_SystemModelObject) {
-                            Fans::SimulateFanComponents(
-                                ZoneEvapUnit(UnitNum).FanName, false, ZoneEvapUnit(UnitNum).FanIndex, _, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
-                        } else {
-                            HVACFan::fanObjs[ZoneEvapUnit(UnitNum).FanIndex]->simulate(_, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
-                        }
-                    }
```

Comment:
This ELSE section was meant to set the flow to zero when the evap unit is off. Now I can do the same with PLR set to zero and running the same function CalcZoneEvapUnitOutput(). This function also set these mass flow rates to zero if the PLR is zero.  I consolidated the On and Off operating modes of the evap unit using the same function CalcZoneEvapUnitOutput() and works for the three ControlSchemeType.